source/ios/AdaptiveCards/AdaptiveCards/AdaptiveCards/ACOFillerSpaceManager.mm

```diff
@@ -0,0 +1,171 @@
+            }
+        }
+    }
+    return NO;
+}
+
+/// configures & adds padding for the `view`
+/// having padding makes the owner of the object,
+/// stretchable
+- (UIView *)addPaddingFor:(UIView *)view
+{
+    if (!view) {
+        return nil;
+    }
+
+    UIView *padding = [[UIView alloc] init];
+
+    [ACOFillerSpaceManager configureHugging:padding];
+
+    NSMutableArray<NSValue *> *values = [_paddingMap objectForKey:view];
+    if (!values) {
+        values = [[NSMutableArray alloc] init];
+        [_paddingMap setObject:values forKey:view];
+    }
+
+    [_stretchableViewSet addObject:padding];
+    [_stretchableViews addObject:[NSValue valueWithNonretainedObject:padding]];
+    [_paddingSet addObject:padding];
+    [values addObject:[NSValue valueWithNonretainedObject:padding]];
+
+    return padding;
+}
+
++ (void)configureHugging:(UIView *)view
+{
+    view.translatesAutoresizingMaskIntoConstraints = NO;
+    [view setContentHuggingPriority:kFillerViewLayoutConstraintPriority forAxis:UILayoutConstraintAxisVertical];
+}
+
+/// configures for AdaptiveCards Height property
+/// Image and Media gets their own padding since stretching them are not desirable
+- (void)configureHeight:(UIView *)view correspondingElement:(ACOBaseCardElement *)correspondingElement
+{
+    if (!view || !correspondingElement || !correspondingElement.element) {
+        return;
+    }
+
+    if ((HeightType::Stretch == correspondingElement.element->GetHeight()) &&
+        (correspondingElement.type != ACRImage &&
+         correspondingElement.type != ACRMedia)) {
+        [ACOFillerSpaceManager configureHugging:view];
+        [_stretchableViewSet addObject:view];
+        [_stretchableViews addObject:[NSValue valueWithNonretainedObject:view]];
+    }
+}
+
+/// activates the constraints together for performance
+/// two stretchable views get same height
+/// by setting low priority, the relationship can be overridden
+/// if it's not possible
+- (NSArray<NSLayoutConstraint *> *)activateConstraintsForPadding
+{
+    if (_stretchableViews.count > 1) {
+        NSMutableArray<NSLayoutConstraint *> *_paddingConstraints = [[NSMutableArray alloc] init];
+        UIView *prevPadding = nil;
+        for (NSValue *paddingValue in _stretchableViews) {
+            UIView *padding = paddingValue.nonretainedObjectValue;
+            if (prevPadding && !padding.isHidden) {
+                [_paddingConstraints addObject:[prevPadding.heightAnchor constraintEqualToAnchor:padding.heightAnchor]];
+                _paddingConstraints.lastObject.priority = UILayoutPriorityDefaultLow;
+            }
+            if (!padding.isHidden) {
+                prevPadding = padding;
+            }
+        }
+
+        if (_paddingConstraints && _paddingConstraints.count) {
+            [NSLayoutConstraint activateConstraints:_paddingConstraints];
+        }
+
+        return _paddingConstraints;
+    }
+    return nil;
+}
+
+- (void)deActivateConstraintsForPadding
```

Comment:
>deActivateConstraintsForPadding [](http://example.com/codeflow?start=8&length=31)

possible rename to deactivateConstraintsForPadding

Reply:
fair enough